src/sentry/replays/usecases/ingest.py

```diff
@@ -55,122 +56,158 @@ class MissingRecordingSegmentHeaders(ValueError):
     pass
 
 
-@metrics.wraps("replays.usecases.ingest.ingest_chunked_recording")
-def ingest_chunked_recording(
+@dataclasses.dataclass
+class RecordingIngestMessage:
+    replay_id: str
+    key_id: int | None
+    org_id: int | None
+    received: int
+    project_id: int
+    payload_with_headers: bytes
+
+
+def ingest_recording_chunked(
     message_dict: RecordingSegmentMessage,
     parts: RecordingSegmentParts,
-    current_transaction: Transaction,
+    transaction: Transaction,
 ) -> None:
-    with current_transaction.start_child(
-        op="replays.process_recording.store_recording", description="store_recording"
+    """Ingest chunked recording messages."""
+    with transaction.start_child(
+        op="replays.usecases.ingest.ingest_recording_chunked",
+        description="ingest_recording_chunked",
     ):
         try:
             recording_segment_with_headers = collate_segment_chunks(parts)
         except ValueError:
             logger.exception("Missing recording-segment.")
             return None
 
-        try:
-            headers, recording_segment = process_headers(recording_segment_with_headers)
-        except MissingRecordingSegmentHeaders:
-            logger.warning(f"missing header on {message_dict['replay_id']}")
-            return None
-
-        with metrics.timer("replays.process_recording.store_recording.count_segments"):
-            count_existing_segments = ReplayRecordingSegmentModel.objects.filter(
-                replay_id=message_dict["replay_id"],
-                project_id=message_dict["project_id"],
-                segment_id=headers["segment_id"],
-            ).count()
-
-        if count_existing_segments > 0:
-            with sentry_sdk.push_scope() as scope:
-                scope.level = "warning"
-                scope.add_attachment(bytes=recording_segment, filename="dup_replay_segment")
-                scope.set_tag("replay_id", message_dict["replay_id"])
-                scope.set_tag("project_id", message_dict["project_id"])
-
-                logging.exception("Recording segment was already processed.")
+        message = RecordingIngestMessage(
+            replay_id=message_dict["replay_id"],
+            key_id=message_dict.get("key_id"),
+            org_id=message_dict.get("org_id"),
+            project_id=message_dict["project_id"],
+            received=message_dict["received"],
+            payload_with_headers=recording_segment_with_headers,
+        )
+        ingest_recording(message, transaction)
 
+        # Segment chunks are always deleted regardless of success or failure.
```

Comment:
I wonder if this is risky to do?

Reply:
Updated the comment's message.  This is an improvement to existing behavior.  If the function does not err then the chunks should be removed because they can not be used in future messages.  If an exception is raised chunks are not cleaned up which mirrors current behavior.